### PR TITLE
chore: UDS based multiproc

### LINKED ISSUE #161: Update multiproc to use UDS instead of TCP
# Summary

As now we use multiple servers instead of one in multiproc, we can transition them to use UDS sockets instead of TCP.
This is also required for container updates as if we use TCP on any server restart the ports are refreshed and the numa container isnt able to reconnect. To solve this we can 
1) Fix the TCP port list
2) Transition to fixed UDS sockets

Number 2 seems more plausible as individually UDS has better performance than TCP

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by kohlisid**
Transition the multiproc servers to use UDS sockets instead of TCP
Fixes #161 

We can see equal CPU usage between the servers
Screenshot 2024-04-24 at 10 33 07 AM

### CODE REVIEW

File: pynumaflow/shared/server.py [hunk 1]
```diff
@@ -92,8 +92,8 @@ def _run_server(
     threads_per_proc,
     server_options,
     udf_type: str,
-    server_info_file,
-    server_info,
+    server_info_file=None,
+    server_info=None,
```
Comment (ab93) [inline pynumaflow/shared/server.py:96]: Do we know the types of these?

Comment (kohlisid) [reply pynumaflow/shared/server.py]: Added

File: pynumaflow/shared/server.py [hunk 2]
```diff
@@ -92,8 +92,8 @@ def _run_server(
     threads_per_proc,
     server_options,
     udf_type: str,
-    server_info_file,
-    server_info,
+    server_info_file: str = None,
```
Comment (ab93) [inline pynumaflow/shared/server.py:95]:
```suggestion
    server_info_file: Optional[str] = None,
```

File: pynumaflow/shared/server.py [hunk 3]
```diff
@@ -92,8 +92,8 @@ def _run_server(
...
+    server_info: ServerInfo = None,
```
Comment (ab93) [inline pynumaflow/shared/server.py:96]:
```suggestion
    server_info: Optional[ServerInfo] = None,
```

**Review by whynowy — APPROVED** [review]

File: pynumaflow/reducestreamer/servicer/task_manager.py [hunk 4]
```diff
@@ -261,7 +261,7 @@ async def process_input_stream(self, request_iterator: AsyncIterable[reduce_pb2.
             # Once all tasks are completed, senf EOF the global result queue
             await self.global_result_queue.put(STREAM_EOF)
         except Exception as e:
-            err_msg = "Reduce Streaming Error: %r" % e.__str__()
+            err_msg = f"Reduce Streaming Error: {str(e)}"
```
Comment (ab93) [inline pynumaflow/reducestreamer/servicer/task_manager.py:264]: Can you do repr here? Gives better information.

File: pynumaflow/_constants.py [hunk 5]
```diff
@@ -16,7 +16,7 @@
 SIDE_INPUT_SOCK_PATH = "/var/run/numaflow/sideinput.sock"
 SOURCE_SOCK_PATH = "/var/run/numaflow/source.sock"
 MULTIPROC_MAP_SOCK_PORT = 55551
```
Comment (vigith) [inline pynumaflow/_constants.py:18]: is this being used?

File: pynumaflow/reducestreamer/servicer/task_manager.py (hunk 4 above)
Comment (kohlisid) [reply pynumaflow/reducestreamer/servicer/task_manager.py]: Done

File: pynumaflow/_constants.py (hunk 5 above)
Comment (kohlisid) [reply pynumaflow/_constants.py]: Removed

**Review by vigith — APPROVED** [review]

**Review by ab93 — APPROVED** [review]

**Comment by ab93** [comment]
Can we do anything about the patch diff here? @kohlisid

**Comment by kohlisid** [comment]
@ab93 That is related to due to the coverage of one exception handler for a condition which is slightly difficult to replicate through UT. I will see if I can add some test there in the other PR

**Pull request merged**